CRM/Core/Permission/Joomla.php

```diff
@@ -53,23 +53,13 @@ public function check($str, $userId = NULL) {
     // not execute hooks if joomla is not loaded
     if (defined('_JEXEC')) {
       $user = JFactory::getUser($userId);
-      $api_key    = CRM_Utils_Request::retrieve('api_key', 'String', $store, FALSE, NULL, 'REQUEST');
+      $api_key = CRM_Utils_Request::retrieve('api_key', 'String');
 
       // If we are coming from REST we don't have a user but we do have the api_key for a user.
       if ($user->id === 0 && !is_null($api_key)) {
-        // This is a codeblock copied from /Civicrm/Utils/REST
-        $uid = NULL;
-        if (!$uid) {
-          $store = NULL;
-
-          $contact_id = CRM_Core_DAO::getFieldValue('CRM_Contact_DAO_Contact', $api_key, 'id', 'api_key');
-
-          if ($contact_id) {
-            $uid = CRM_Core_BAO_UFMatch::getUFId($contact_id);
-          }
-          $user = JFactory::getUser($uid);
-
-        }
+        $contact_id = CRM_Core_DAO::getFieldValue('CRM_Contact_DAO_Contact', $api_key, 'id', 'api_key');
+        $uid = ($contact_id) ? CRM_Core_BAO_UFMatch::getUFId($contact_id) : NULL;
+        $user = JFactory::getUser($uid);
```

Comment:
I can't recall exactly but i think getUser might not like NULLs

Reply:
Looks OK: https://api.joomla.org/cms-2.5/classes/JFactory.html#method_getUser
In any case, no change here - we were passing NULL before if the contact_id was not found.